### ISSUE TITLE
spyder crashes/hangs repeatedly

### ISSUE THREAD

**Issue by mczakk** [spyder-ide/spyder#20867]
### Issue Report Checklist

* [ x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ x] Tried basic troubleshooting (if a bug/error)
    * [ x] Restarted Spyder
    * [ x] Reset preferences with ``spyder --reset``
    * [ x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x ] Tried the other applicable steps from the Troubleshooting Guide
* [ x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
when running any script spyder will just hang after producing the runfile ('xxxx', wdir='xxx', current_namespace=True) message in the console.
I can still enter type into the editor, but clicking the red square int the console does nothing.
I can click on restart kernel, sometimes this will allow the script to run, sometimes not.
generally I have to close and re-open spyder, but this does not always work.

### What steps reproduce the problem?

1. Running any script
2. 
3. 

### What is the expected output? What do you see instead?
nothing, it is currently hung, and I have copied the spyder version info and dependencies while it is 'running'.
attached is a screenshot of the console
![spyderbad](https://user-images.githubusercontent.com/6179722/234060266-f98bf07d-b847-46b6-84fc-6630d37d425a.jpg)

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions
* Spyder version: 5.4.3  (conda)
* Python version: 3.11.3 64-bit
* Qt version: 5.15.6
* PyQt5 version: 5.15.7
* Operating System: Windows 10

* Spyder version: 
* Python version: 
* Qt version: 
* PyQt version: 
* Operating System name/version: 

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                             :  1.4.1 (OK)
chardet >=2.0.0                                  :  5.1.0 (OK)
cloudpickle >=0.5.0                              :  2.2.1 (OK)
cookiecutter >=1.6.0                             :  2.1.1 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.0.2 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.12.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.2 (OK)
jellyfish >=0.7                                  :  0.9.0 (OK)
jsonschema >=3.2.0                               :  4.17.3 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  7.3.1 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
paramiko >=2.4.0                                 :  3.1.0 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.4 (OK)
pygments >=2.0                                   :  2.15.0 (OK)
pylint >=2.5.0,<3.0                              :  2.17.2 (OK)
pylint_venv >=2.1.1                              :  3.0.1 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  1.2.1 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.1 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.3 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.3.1 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  67.6.1 (OK)
sphinx >=0.6.6                                   :  6.1.3 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.3 (OK)
textdistance >=4.2.0                             :  4.5.0 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  3.0.0 (OK)
zmq >=22.1.0                                     :  25.0.2 (OK)

# Optional:
cython >=0.21                                    :  0.29.34 (OK)
matplotlib >=3.0.0                               :  3.7.1 (OK)
numpy >=1.7                                      :  1.24.2 (OK)
pandas >=1.1.1                                   :  2.0.0 (OK)
scipy >=0.17.0                                   :  1.10.1 (OK)
sympy >=0.7.3                                    :  1.11.1 (OK)

```

**Comment by ccordoba12** [comment]
Hey @mczakk, thanks for reporting. What Matplotlib backend are you using to generate your plots? It's ok if you don't know what that is, but that could be the cause of your issues.

**Comment by mczakk** [comment]
> Ok, then please open a system terminal (i.e. cmd.exe) and run there
> 
> ```
> spyder --reset
> ```
> 
> and try again. Perhaps that solves the problem for you.

Not sure what this will achieve, as I said, I have uninstalled Anaconda and it's associated version of Spyder, and installed standalone spyder using pip. 
This seems to be working ok :)

**Comment by gica3618** [comment]
I was having the same problem. I uninstalled and re-installed Anaconda, but no success. However, running `spyder --reset` fixed it!

**Comment by gica3618** [comment]
Celebrated too soon... When I activate the option "Remove all variables before execution" in the settings, it hangs again. Disabling the option makes it work again. Is there a solution to this problem?

**Comment by ccordoba12** [comment]
Thanks for the extra info @gica3618! This means that this is a duplicate of issue #16138 and we'll try to fix it in Spyder 6, to be released during the second half of the year.

**Comment by ccordoba12** [comment]
For now, you can't use that option because it's broken, sorry.

**Comment by ccordoba12** [comment]
@K4-z3, that was a problem with older Spyder versions. If you're using the latest one (5.4.3), you shouldn't see it.

**Comment by K4-z3** [comment]
> @K4-z3, that was a problem with older Spyder versions. If you're using the latest one (5.4.3), you shouldn't see it.

I'm currently using Spyder 5.4.3 with Python 3.8.16, Qt 5.15.2 and PyQt5 5.15.7 and apparently still got the issue.

**Comment by ccordoba12** [comment]
@K4-z3, please open a new issue about your problem and don't forget to fill all the info required by our issue template.

**Comment by DiegoRodasRodriguez** [comment]
Hi @ccordoba12 I saw you mentioned that the issue should be fixed for Spyder v5.4.3 and above, but I am currently using (5.51) with Python 3.12.4 64-bit | Qt 5.15.2 | PyQt5 5.15.10 and I have the same problem about Qt5. If my backend is set to Qt5 my console freezes and if I change to Automatic or Inline it seems to work properly.

**Comment by ccordoba12** [comment]
@DiegoRodasRodriguez, that problem comes and goes due to changes in some Jupyter packages (yes, we use the same execution architecture as Jupyter).

Please open the Anaconda Prompt and run these commands there:

    conda create -n spyder-env -c conda-forge spyder=5.5.6 matplotlib=3.8
    conda activate spyder-env
    spyder

That Spyder version should work as expected.

**Comment by lagunabravo** [comment]
I came to this thread as I kept getting repeated freezes on first launching Spyder 5.5.6 from Anaconda in Windows. It only happens after first starting the computer, when Spyder is idle and, noticeably, after about 60 seconds. So, in case it helps anyone, it my case I believe it is related to the update suggestion window. It resolved when I managed to dismiss update suggestion the one time it popped while I had the preferences window opened. 

 I do have other versions of Spyder including 6 installed outside Anaconda and that could be interfering with the suggestion window.
